scipy/fft/_helper.py

```diff
@@ -98,4 +101,192 @@ def _init_nd_shape_and_axes(x, shape, axes):
         Axes along which the calculation is computed. It is a 1-D integer array.
 
     """
-    return _helper._init_nd_shape_and_axes(x, shape, axes)
+    xp = array_namespace(x)
+    x = np.asarray(x)
+    shape, axes = _helper._init_nd_shape_and_axes(x, shape, axes)
+    return xp.asarray(shape), xp.asarray(axes)
+
+
+def fftfreq(n, d=1.0, *, xp=None, device=None):
+    """Return the Discrete Fourier Transform sample frequencies.
+
+    The returned float array `f` contains the frequency bin centers in cycles
+    per unit of the sample spacing (with zero at the start).  For instance, if
+    the sample spacing is in seconds, then the frequency unit is cycles/second.
+
+    Given a window length `n` and a sample spacing `d`::
+
+      f = [0, 1, ...,   n/2-1,     -n/2, ..., -1] / (d*n)   if n is even
+      f = [0, 1, ..., (n-1)/2, -(n-1)/2, ..., -1] / (d*n)   if n is odd
+
+    Parameters
+    ----------
+    n : int
+        Window length.
+    d : scalar, optional
+        Sample spacing (inverse of the sampling rate). Defaults to 1.
+
+    Returns
+    -------
+    f : ndarray
+        Array of length `n` containing the sample frequencies.
+
+    Examples
+    --------
+    >>> signal = np.array([-2, 8, 6, 4, 1, 0, 3, 5], dtype=float)
+    >>> fourier = np.fft.fft(signal)
+    >>> n = signal.size
+    >>> timestep = 0.1
+    >>> freq = np.fft.fftfreq(n, d=timestep)
```

Comment:
Since we have our own docstring we can call `scipy.fft.fftfreq` now.

Reply:
Yep this is planned for a follow-up.

Reply:
This has been addressed in https://github.com/scipy/scipy/pull/19005/commits/8994857d513021a65a35c3f7d040f09eb5c4bd33 now.